com.unity.netcode.gameobjects/Runtime/Transports/UTP/UnityTransport.cs

```diff
@@ -1432,6 +1489,42 @@ private void ConfigureSimulatorForUtp1()
 #endif
                 heartbeatTimeoutMS: transport.m_HeartbeatTimeoutMS);
 
+#if UTP_TRANSPORT_2_0_ABOVE
+            if (m_UseEncryption)
+            {
+                try
+                {
+                    if (NetworkManager.IsServer)
+                    {
+#if UNITY_WEBGL && !UNITY_EDITOR
+                        throw new Exception("WebGL as a server is not supported by Unity Transport, outside the Editor.");
+#else
+                        if (m_ServerCertificate.Length == 0 ||
+                            m_ServerPrivate.Length == 0)
+                        {
+                            throw new Exception("In order to use encrypted communications, when hosting, you must set the server certificate and key.");
+                        }
+                        m_NetworkSettings.WithSecureServerParameters( certificate: ref m_ServerCertificate,
+                            privateKey: ref m_ServerPrivate);
+#endif
+                    }
+                    else
+                    {
+                        if (m_ServerCommonName.Length == 0 ||
+                            m_ClientCertificate.Length == 0)
+                        {
+                            throw new Exception("In order to use encrypted communications, clients must set the server common name and client certificate.");
```

Comment:
That's not the case if using certificates from a "real" CA provider (where we want the server certificate to be validated against the OS/browser database). In this scenario, not providing the CA certificate is the expected workflow (there's a version of `WithSecureClientParameters` that only takes the server name as a parameter).

Reply:
Yes, very valid point, thanks. I'll remove the check on `m_ClientCertificate`.